lib/carrierwave/storage/aws_file.rb

```diff
@@ -58,7 +58,10 @@ def store(new_file)
         if new_file.is_a?(self.class)
           new_file.move_to(path)
         else
-          file.upload_file(new_file.path, aws_options.write_options(new_file))
+          options = aws_options.write_options(new_file).except(:body)
+          options[:multipart_threshold] = AWSOptions::MULTIPART_THRESHOLD
+          Aws::S3::TransferManager.new(client: connection.client).upload_file(new_file.path, bucket: bucket.name,
+                                                                                             key: path, **options)
```

Comment:
`Aws::S3::TransferManager` was introduced in aws-sdk-s3 version 1.197.0.
To keep compatibility with earlier versions, it'd be better to implement it like this:

```suggestion
          if Aws::S3.const_defined?(:TransferManager)
            options = aws_options.write_options(new_file).except(:body)
            options[:multipart_threshold] = AWSOptions::MULTIPART_THRESHOLD
            Aws::S3::TransferManager.new(client: connection.client).upload_file(new_file.path, bucket: bucket.name,
                                                                                               key: path, **options)
          else
            file.upload_file(new_file.path, aws_options.write_options(new_file))
          end
```